07.ls_object/outputter.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require_relative 'file_detail'
+
+class Outputter
+  COLUMNS = 3
+
+  def initialize(file_names)
+    @file_details = file_names.map { |file_name| FileDetail.new(file_name) }
+  end
+
+  def output(long: false)
+    long ? long_output : no_option_output
+  end
+
+  private
+
+  def long_output
+    puts "total #{total_blocks}"
+    @file_details.each do |file_detail|
+      puts long_format(file_detail)
+    end
+  end
+
+  def no_option_output
+    formatted_files.each do |files|
+      files.compact.each { |file| print file.name.ljust(30) }
+      puts "\n"
+    end
+  end
+
+  def total_blocks
+    @file_details.sum { |file_detail| file_detail.blocks }
```

Comment:
```suggestion
    @file_details.sum(&:blocks)
```

だけで済みます。これだけならメソッド定義をなくしちゃってもいいかもしれません。